mnist/README.md

```diff
@@ -6,165 +6,96 @@ This example guides you through the process of taking an example model, modifyin
 
 Before we get started there a few requirements.
 
-### Kubernetes Cluster Environment
+## 1.Kubernetes Cluster Environment
 
-Your cluster must:
+For supported versions, please check Kubeflow document: https://www.kubeflow.org/docs/started/getting-started/
 
-- Be at least version 1.9
-- Have access to an S3-compatible object store ([Amazon S3](https://aws.amazon.com/s3/), [Google Storage](https://cloud.google.com/storage/docs/interoperability), [Minio](https://www.minio.io/kubernetes.html))
-- Contain 3 nodes of at least 8 cores and 16 GB of RAM.
-
-If using GKE, the following will provision a cluster with the required features:
-
-```
-export CLOUDSDK_CONTAINER_USE_CLIENT_CERTIFICATE=True
-gcloud alpha container clusters create ${USER} --enable-kubernetes-alpha --machine-type=n1-standard-8 --num-nodes=3 --disk-size=200 --zone=us-west1-a --cluster-version=1.9.3-gke.0 --image-type=UBUNTU
-```
-
-If using Azure, the following will provision a cluster with the required features, [using the az cli](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest):
+## 2.ksonnect cli
 
 ```
-# Create a resource group
-az group create -n kubeflowrg -l eastus
-# Deploy the cluster
-az aks create -n kubeflowaks -g kubeflowrg -l eastus -k 1.9.6 -c 3 -s Standard_NC6
-# Authentication into the cluster
-az aks get-credentials -n kubeflowaks -g kubeflowrg
+curl -fksSL https://github.com/ksonnet/ksonnet/releases/download/v0.13.0/ks_0.13.0_linux_amd64.tar.gz \
+    | tar --strip-components=1 -xvz -C /usr/local/bin/ ks_0.13.0_linux_amd64/ks
 ```
 
-NOTE: You must be a Kubernetes admin to follow this guide. If you are not an admin, please contact your local cluster administrator for a client cert, or credentials to pass into the following commands:
+## 3.Kubeflow environment
 
 ```
-$ kubectl config set-credentials <username> --username=<admin_username> --password=<admin_password>
-$ kubectl config set-context <context_name> --cluster=<cluster_name> --user=<username> --namespace=<namespace>
-$ kubectl config use-context <context_name>
+git clone -b v0.3.2 https://github.com/kubeflow/kubeflow
+./kubeflow/scripts/kfctl.sh init kf-app --platform none
+cd kf-app
+../kubeflow/scripts/kfctl.sh generate k8s
+../kubeflow/scripts/kfctl.sh apply k8s
 ```
 
-### Local Setup
+For detail info, check here: https://www.kubeflow.org/docs/started/getting-started/
 
-You also need the following command line tools:
+Note: The vizier-db in Kubeflow depends on a PV, you need to create it in order to make vizier-db work. But this does not affect this example.
 
-- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
-- [argo](https://github.com/argoproj/argo/blob/master/demo.md#1-download-argo)
-- [ksonnet](https://ksonnet.io/#get-started)
 
-To run the client at the end of the example, you must have [requirements.txt](requirements.txt) intalled in your active python environment.
+## 4.Setup Minio
```

Comment:
I feel that recommending minio (while it is an excellent option) as the default setup is not setting the right tone for pragmatic engineering. Unless there are specific requirements, or your environment has no S3-compatible object store, there isn't a need for Minio. It just becomes one more moving piece in an already complicated example.

Reply:
+1 to @elsonrodriguez to not recommending minio for beginners. That seems like a distraction. If users don't have a Cloud account they are probably running on minikube/microk8s/docker. So I think the best option is to just use a PV even if that means running non-distributed.